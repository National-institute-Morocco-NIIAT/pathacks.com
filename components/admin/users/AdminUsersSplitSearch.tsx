import React from 'react'
import { splitWithSearchTerm } from '../../../helpers/splitWithSearchTerm'

export const AdminUsersSplitSearch = (str: string, searchTerm: string) => {
  // make all lowercase now to ensure both lower and uppercase characters can be searched for
  const lowerCaseSearchTerm = searchTerm.toLowerCase()

  // convert string to array with searchTerm included
  const splitArr = splitWithSearchTerm(str, searchTerm)

  // highlight search Term
  const res = splitArr.map((word: string, key: number) => {
    const bgColor =
      word.toLowerCase() === lowerCaseSearchTerm ? 'rgb(84, 64, 216, .25)' : ''
    return (
      <span key={word + searchTerm + key} style={{ backgroundColor: bgColor }}>
        {word}
      </span>
    )
  })

  return res
}
