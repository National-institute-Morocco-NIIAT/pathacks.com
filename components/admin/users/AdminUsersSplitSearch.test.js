import { render } from '@testing-library/react'
import { AdminUsersSplitSearch } from './AdminUsersSplitSearch'

describe('AdminUsersSplitSearch component', () => {
  test('Should highlight full search hit', async () => {
    const { container } = render(AdminUsersSplitSearch('newbie', 'username'))
    expect(container).toMatchSnapshot()
  })
  test('Should highlight partial search hit', async () => {
    const { container } = render(AdminUsersSplitSearch('nameManeName', 'name'))
    expect(container).toMatchSnapshot()
  })
})
