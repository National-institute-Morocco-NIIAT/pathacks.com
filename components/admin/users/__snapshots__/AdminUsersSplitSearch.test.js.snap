// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminUsersSplitSearch component Should highlight full search hit 1`] = `
<div>
  <span>
    newbie
  </span>
</div>
`;

exports[`AdminUsersSplitSearch component Should highlight partial search hit 1`] = `
<div>
  <span>
    name
  </span>
  <span>
    Mane
  </span>
  <span>
    Name
  </span>
</div>
`;
